### PR TITLE
233 Create page heading component

### PR DESCRIPTION
**Pull request by SamJamCul**
### Context
https://trello.com/c/dbEPQ9Bo/233-m-update-page-titles-to-all-pages

One issue is for pages that don't have a clear `h1` tag on the page don't have a clear title that would be suitable - an example being the /trainees/1/course-details page.

### Changes proposed in this pull request
* Add page heading component, and replace h1 headings on pages where appropriate
* Add new locales info for each page heading

### CODE REVIEW

**Comment by SamJamCul** [comment]
[[M] Update page titles to all pages](https://trello.com/c/dbEPQ9Bo/233-m-update-page-titles-to-all-pages)

File: app/views/trainees/new.html.erb [hunk 1]
```diff
@@ -11,9 +11,8 @@
 
       <%= f.govuk_error_summary %>
 
-      <h1 class="govuk-heading-l">
-        Add a trainee
-      </h1>
+      <%= render_component PageHeading::View.new(title: "trainees.new", 
+                                                 errors: @trainee.errors) %>
```
Comment (edwardhorsford) [inline app/views/trainees/new.html.erb:15]: I'm not sure why this would need to know about errors - errors have no bearing on the page headings.

File: app/components/page_heading/view.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    @errors = errors.present?
+  end
+
+  def build_page_title
+    [build_error + build_title, build_service_name, "GOV.UK"].join(" - ")   
+  end
+
+  def build_page_heading
+    build_title
+  end
```
Comment (edwardhorsford) [inline app/components/page_heading/view.rb:15]: I find this hard to follow - it's a bit circular. 

`build_page_heading` calls `build_title` that then calls `page_headings` 

I suspect page heading should just use whatever page heading is defined int the translation file - there's nothing to build.

Page titles should then either use that, or define a new one.

File: spec/components/page_heading/view_preview.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require "govuk/components"
+class PageHeading::ViewPreview < ViewComponent::Preview
+  def default_heading
+    render_component(PageHeading::View.new(title: "trainees.new"))
+  end
+
+  def heading_with_error
+    render_component(PageHeading::View.new(title: "trainees.new", errors: "Error"))
+  end
```
Comment (edwardhorsford) [inline spec/components/page_heading/view_preview.rb:9]: Should this be `title_with_error`?

File: app/components/page_heading/view.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  attr_accessor :title, :errors
+  include ActiveModel
+  def initialize(title: "", errors: false)
+    @title = title
+    @errors = errors.present?
+  end
+
+  def build_page_title
+    [build_error + build_title, build_service_name, "GOV.UK"].join(" - ")   
+  end
```
Comment (edwardhorsford) [inline app/components/page_heading/view.rb:11]: Apply's version:
https://github.com/DFE-Digital/apply-for-teacher-training/blob/9227dcb8689bdd9390aecafce7f3b51510cb3189/app/helpers/application_helper.rb#L2

looks like it handles not having a title (which we probably want on the homepage/start page.

File: app/components/page_heading/view.rb (hunk 2 above)
Comment (SamJamCul) [reply app/components/page_heading/view.rb]: I've changed it so that the page title component now draws from its own section of the locales now.

File: app/components/page_heading/view.rb (hunk 4 above)
Comment (SamJamCul) [reply app/components/page_heading/view.rb]: Should be good for now, it can handle there being no title, in which case it goes to the default of not using it.

File: app/components/page_heading/view.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+class PageHeading::View < GovukComponent::Base
+  attr_accessor :title, :errors
+  include ActiveModel
+  def initialize(title: "", errors: false)
+    @title = title
+    @errors = errors.present?
+  end
+
+  def build_page_title
```
Comment (darokel) [inline app/components/page_heading/view.rb:9]: Could we have a spec for this component?

File: spec/components/page_heading/view_preview.rb (hunk 3 above)
Comment (SamJamCul) [reply spec/components/page_heading/view_preview.rb]: Sorted

File: app/components/page_heading/view.rb (hunk 5 above)
Comment (SamJamCul) [reply app/components/page_heading/view.rb]: Done!

**Review by aliuk2012 — APPROVED** [review]

File: app/components/page_title/view.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
+class PageTitle::View < GovukComponent::Base
+  attr_accessor :title, :errors
+  include ActiveModel
```
Comment (gpeng) [inline app/components/page_title/view.rb:3]: Can we put some whitespace in 'ere?

File: spec/components/page_title/view_preview.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+require "govuk/components"
```
Comment (gpeng) [inline spec/components/page_title/view_preview.rb:1]: and 'ere.

**Review by gpeng — APPROVED** [review]

File: spec/components/page_title/view_preview.rb (hunk 7 above)
Comment (SamJamCul) [reply spec/components/page_title/view_preview.rb:1]: Donezo
